example/z-docker/Dockerfile

```diff
@@ -1,4 +1,40 @@
-FROM ubuntu:latest
-RUN apt update && apt install -y openssl libev4
-COPY app.exe .
-ENTRYPOINT /app.exe
+FROM ocaml/opam:alpine as build
+
+# Install system dependencies
+# Some of these are only needed for database and JavaScript stuff, but this is 
+# our build context, it won't impact our runtime environment.
+RUN sudo apk add --update \
+    make libffi-dev linux-headers m4 ncurses-dev perl pkgconf postgresql-dev \
+    musl-dev git pkgconfig gmp-dev libc-dev openblas-dev zlib-dev \
+    npm nodejs libev-dev 
+
+# Install Esy-specific dependencies
+RUN sudo apk add \
+    perl-utils
+
+RUN eval $(opam env) && opam repository set-url default https://opam.ocaml.org && opam update
+
+WORKDIR /build
+
+RUN npm install esy
+
+# Install dependencies
+# We only copy the opam file to install dependencies only when they change, and 
+# not every time another file is modified.
```

Comment:
This comment is out of sync with the build system being used here. Can you customize this file to `esy`?

Reply:
Done